src/main/java/frc/robot/subsystems/led/LedMatrixSubsystem.java

```diff
@@ -0,0 +1,139 @@
+package frc.robot.subsystems.led;
+
+import edu.wpi.first.wpilibj.util.Color;
+import edu.wpi.first.wpilibj2.command.SubsystemBase;
+
+import java.util.Arrays;
+import java.util.Collections;
+
+import edu.wpi.first.networktables.GenericEntry;
+import edu.wpi.first.wpilibj.AddressableLED;
+import edu.wpi.first.wpilibj.AddressableLEDBuffer;
+import edu.wpi.first.wpilibj.shuffleboard.BuiltInWidgets;
+import edu.wpi.first.wpilibj.shuffleboard.Shuffleboard;
+import edu.wpi.first.wpilibj.shuffleboard.ShuffleboardTab;
+import frc.robot.hardware.LedDriver;
+
+public class LedMatrixSubsystem extends SubsystemBase{
+    private ShuffleboardTab ledTab = Shuffleboard.getTab("Led");
+    private GenericEntry ledBrightnessSlider = ledTab.add("Brightness", 0.2).withWidget(BuiltInWidgets.kNumberSlider).getEntry();
+    String[][] colors = LedDriver.gocans;
+    String[][] gocans = LedDriver.gocans;
+    String[][] one = LedDriver.one;
+    String[][] two = LedDriver.two;
+    String[][] three = LedDriver.three;
```

Comment:
Since the arrays are already public, no reason to make direct copies here. Maybe if you used the helper method I suggested in another comment, you could make convenience copies like this:
```private final String[][] gocans = serpentine(LedDriver.gocans);```